Clustering: panic: send on closed channel

Have seen this panic running tests on Windows:
```
=== RUN   TestRAFTTransportConnReader/#02
panic: send on closed channel
goroutine 38782 [running]:
github.com/nats-io/nats-streaming-server/server.(*natsConn).onMsg(0xc001beaa00, 0xc008740d70)
	C:/Users/travis/gopath/src/github.com/nats-io/nats-streaming-server/server/raft_transport.go:115 +0x10e
github.com/nats-io/nats-streaming-server/vendor/github.com/nats-io/nats%2ego.(*Conn).waitForMsgs(0xc0030d3b80, 0xc00ad21980)
	C:/Users/travis/gopath/src/github.com/nats-io/nats-streaming-server/vendor/github.com/nats-io/nats.go/nats.go:2211 +0x352
created by github.com/nats-io/nats-streaming-server/vendor/github.com/nats-io/nats%2ego.(*Conn).subscribeLocked
	C:/Users/travis/gopath/src/github.com/nats-io/nats-streaming-server/vendor/github.com/nats-io/nats.go/nats.go:3048 +0x494
```

--- a/server/raft_transport.go
+++ b/server/raft_transport.go
@@ -100,22 +100,20 @@ func (n *natsConn) onMsg(msg *nats.Msg) {
 		return
 	}
 	pb.buf = msg.Data
-	var notify bool
 	if n.pendingT != nil {
 		n.pendingT.next = pb
 	} else {
 		n.pendingH = pb
-		notify = true
-	}
-	n.pendingT = pb
-	n.mu.Unlock()
-
-	if notify {
+		// Need to notify under the lock since this channel gets closed
+		// on natsConn.close(), which could then cause a panic of
+		// "send on closed channel".
 		select {
 		case n.ch <- struct{}{}:
 		default:
 		}
 	}
+	n.pendingT = pb
+	n.mu.Unlock()
 }
 
 func (n *natsConn) Read(b []byte) (int, error) {

